addons/udes_delivery_control/models/stock_picking.py

```diff
@@ -39,38 +39,43 @@ class StockPicking(models.Model):
         related="picking_type_id.u_is_delivery_control", readonly=1
     )
     u_extras_id = fields.Many2one("stock.picking.extras", copy=False)
-    # Unloading Fields
-    u_user_id = fields.Many2one("res.users", related="u_extras_id.user_id",)
-    u_location_id = fields.Many2one("stock.location", related="u_extras_id.location_id")
-    u_vehicle_arrival_date = fields.Datetime(related="u_extras_id.vehicle_arrival_date",)
+    u_loading_type = fields.Selection(related="u_extras_id.loading_type", required=True)
+
+    # Common fields
+    u_user_id = fields.Many2one("res.users", related="u_extras_id.user_id")
+    u_vehicle_arrival_date = fields.Datetime(related="u_extras_id.vehicle_arrival_date")
     u_week = fields.Integer(related="u_extras_id.week")
-    u_unloading_start_date = fields.Datetime(related="u_extras_id.unloading_start_date")
-    u_unloading_end_date = fields.Datetime(related="u_extras_id.unloading_end_date")
-    u_unloading_time_taken = fields.Float(readonly=True, related="u_extras_id.unloading_time_taken")
     u_vehicle_type = fields.Many2one(
         "stock.picking.vehicle.type", related="u_extras_id.vehicle_type"
     )
     u_trailer_number = fields.Integer(related="u_extras_id.trailer_number")
+    u_is_planned = fields.Boolean(related="u_extras_id.is_planned")
+    u_is_late = fields.Boolean(related="u_extras_id.is_late")
+    u_is_fast_track = fields.Boolean(related="u_extras_id.is_fast_track")
+
+    # Unloading Fields
+    u_is_unload = fields.Boolean(related="u_extras_id.is_unload")
+    u_location_id = fields.Many2one("stock.location", related="u_extras_id.location_id")
+    u_unloading_start_date = fields.Datetime(related="u_extras_id.unloading_start_date")
+    u_unloading_end_date = fields.Datetime(related="u_extras_id.unloading_end_date")
+    u_unloading_time_taken = fields.Float(readonly=True, related="u_extras_id.unloading_time_taken")
     u_lane_number = fields.Char(related="u_extras_id.lane_number")
     u_pallet_count = fields.Integer(related="u_extras_id.pallet_count")
     u_stillage_count = fields.Integer(related="u_extras_id.stillage_count")
     u_box_count = fields.Integer(related="u_extras_id.box_count")
-    u_is_planned = fields.Boolean(related="u_extras_id.is_planned",)
-    u_is_late = fields.Boolean(related="u_extras_id.is_late",)
-    u_is_fast_track = fields.Boolean(related="u_extras_id.is_fast_track",)
 
     # Backloading Fields
-    u_is_backload = fields.Boolean(related="u_extras_id.is_backload",)
+    u_is_backload = fields.Boolean(related="u_extras_id.is_backload")
```

Comment:
Do we need `u_is_backload` and `u_is_unload`? Can't we just check `loading_type` for any checks?

Reply:
I suppose it makes it easier to just check a boolean value rather than having to check if a selection value is in a list of values.